xwiki-platform-core/xwiki-platform-rendering/xwiki-platform-rendering-macros/xwiki-platform-rendering-macro-message/src/main/resources/META-INF/components.txt

```diff
@@ -0,0 +1,4 @@
+org.xwiki.rendering.internal.macro.message.XWikiInfoMessageMacro
+org.xwiki.rendering.internal.macro.message.XWikiWarningMessageMacro
+org.xwiki.rendering.internal.macro.message.XWikiErrorMessageMacro
+org.xwiki.rendering.internal.macro.message.XWikiSuccessMessageMacro
```

Comment:
I think these components need to be declared with a lower priority (I think we typically use 500) to actually reliably override the components from `xwiki-rendering`. Further, the module needs to be added as a dependency somewhere, the most relevant place probably being https://github.com/xwiki/xwiki-platform/blob/3d4b2e28553b594b1ddb255de69c3b645b551e54/xwiki-platform-distribution/xwiki-platform-distribution-war-dependencies/pom.xml#L301-L306. Now we have lots of modules that also directly depend on the macros from `xwiki-rendering`, I don't know if we should all change them.

Reply:
Yes, any component designed to overwrite another should be prefixed with `500:` (or something else for more specific cases).

Reply:
> I think these components need to be declared with a lower priority (I think we typically use 500) to actually reliably override the components from xwiki-rendering. 

Added the priority in https://github.com/xwiki/xwiki-platform/pull/3023/commits/9e7759387dc99ee81b038d9712712fd62ab6ada7 :+1: . 


> Further, the module needs to be added as a dependency somewhere, the most relevant place probably being [...]

I replaced the xwiki-rendering dependency here with the overriding module.
However I don't understand too well why we'd need to do it only here and in none of the other results of:
https://github.com/search?q=repo%3Axwiki%2Fxwiki-platform%20xwiki-rendering-macro-message&type=code

Reply:
If you don't change the other places, in these other places, you won't have the new macros. For an actual installation, the dependency in the war will be installed, and thus the macros are overridden. Most of these UI modules have an extra dependency to have the macro available in the page tests. In these page tests, the original macro will thus be used and not the overridden one.